<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <joint name="lidar_joint" type="fixed">
        <parent link="chassis"/>
        <child link="lidar_link"/>
        <origin xyz="0.15 0 0.15" rpy="0 0 0"/>
    </joint>

    <link name="lidar_link">
        <visual>
            <geometry>
                <cylinder radius="0.05" length="0.04"/>
            </geometry>
            <material name="red"/>
        </visual>
        <collision>
            <geometry>
                <cylinder radius="0.05" length="0.04"/>
            </geometry>
        </collision>
        <xacro:inertial_cylinder mass="0.1" length="0.04" radius="0.05">
            <origin xyz="0 0 0" rpy="0 0 0"/>
        </xacro:inertial_cylinder>
    </link>

    <gazebo reference="lidar_link">
        <material>Gazebo/Red</material>
        <!-- <sensor type="ray" name="lidar_sensor">
            <pose>0 0 0 0 0 0</pose>
            <visualize>true</visualize>
            <update_rate>10</update_rate>
            <ray>
                <scan>
                    <horizontal>
                        <samples>720</samples>
                        <resolution>1</resolution>
                        <min_angle>-3.14159</min_angle>
                        <max_angle>3.14159</max_angle>
                    </horizontal>
                </scan>
                <range>
                    <min>0.10</min>
                    <max>12.0</max>
                    <resolution>0.01</resolution>
                </range>
            </ray>

            <plugin name="gazebo_ros_ray_sensor" filename="libgazebo_ros_ray_sensor.so">
                <ros>
                    <namespace>/demo</namespace>
                    <output_type>sensor_msgs/LaserScan</output_type>
                    <topic>scan</topic>
                    <frame_name>lidar_link</frame_name>
                </ros>
            </plugin>
        </sensor> -->
    </gazebo>

</robot>